### ISSUE TITLE
Feature support for trustedImages

### ISSUE THREAD

**Issue by luhaixun** [hadolint/hadolint#357]
- [] This is a bug report
- [x] This is a feature request
- [x] I searched existing issues before opening this one

### Expected behavior
Hadolint can support trusted images. That being said, only the trusted images can be the base image inside the **FROM** keyword

### Actual behavior
Hadolint supports trusted registries but doesn't support trusted images

**Use trustedImages**
If the below section is configured in the hadolint YAML file, Hadolint will only allow below 2 configured images could be the base images in Dockerfile.

```yaml
trustedImages:
  - haskell:8.8.1
  - hadolint/hadolint:v1.17.2
```

**Comment by ghostsquad** [comment]
Similar to #401 #446

**Comment by joanayma** [comment]
We will need some similar feature to limit the `docker.io` registry to i.g. `docker.io/library/*`. Is there any plan or work-in-progress? Thanks!
